Model/Api/Product.php

```diff
@@ -340,18 +340,23 @@ protected function _buildProductData(
     ) {
 
         $data = [];
+        $parent = null;
 
         //data applied for both root and varient products
         $data["id"] = $product->getId();
         $data["title"] = $product->getName();
         $data["url"] = $product->getProductUrl();
-        if ($product->getImage()) {
+        if ($product->getImage()&&$product->getImage()!='no_selection') {
             $filePath = 'catalog/product'.$product->getImage();
             $data["image_url"] = $this->_helper->getBaserUrl($magentoStoreId, \Magento\Framework\UrlInterface::URL_TYPE_MEDIA).$filePath;
         } elseif ($this->_parentImage) {
```

Comment:
$this->_parentImage should be removed or the logic should be changed, as line 431 could result in empty image in some case. 

if (isset($data['image_url'])) {
  $this->_parentImage = $data['image_url'];
}

OR the condition on this should be changed to use empty function instead of issue